crates/primitives/src/bloom.rs

```diff
@@ -1,10 +1,61 @@
 //! Bloom related utilities.
+use crate::{keccak256, Log};
+use bytes::Buf;
+use derive_more::{AsRef, Deref};
+use fixed_hash::construct_fixed_hash;
+use impl_serde::impl_fixed_hash_serde;
+use reth_codecs::{impl_hash_compact, Compact};
 
-use crate::{keccak256, Bloom, Log};
+#[cfg(test)]
+use proptest::{
+    arbitrary::{any_with, Arbitrary as PropTestArbitrary, ParamsFor},
+    strategy::{BoxedStrategy, Strategy},
+};
+
+#[cfg(any(test, feature = "arbitrary"))]
+use arbitrary::Arbitrary;
 
 /// Length of bloom filter used for Ethereum.
 pub const BLOOM_BYTE_LENGTH: usize = 256;
 
+construct_fixed_hash! {
+    /// 2048 bits type.
+    #[cfg_attr(any(test, feature = "arbitrary"), derive(Arbitrary))]
+    #[derive(AsRef, Deref)]
+    pub struct Bloom(BLOOM_BYTE_LENGTH);
+}
+
+impl_hash_compact!(Bloom);
+impl_fixed_hash_serde!(Bloom, BLOOM_BYTE_LENGTH);
+
+#[cfg(test)]
+impl PropTestArbitrary for Bloom {
```

Comment:
Would've been nice if this could be implemented for anything that implements `From<u8>` so we didnt need to write this manually